crates/nu-protocol/src/errors/shell_error/io.rs

```diff
@@ -369,16 +385,15 @@ impl Diagnostic for IoError {
             (true, Some(location)) => SourceSpan::new(0.into(), location.len()),
         };
 
-        let label = match self.additional_context.as_ref() {
-            Some(ctx) => format!("{ctx}\n{}", self.kind),
-            None => self.kind.to_string(),
-        };
-        let label = LabeledSpan::new_with_span(Some(label), span);
+        let label = LabeledSpan::new_with_span(Some(self.kind.to_string()), span);
         Some(Box::new(std::iter::once(label)))
     }
 
     fn diagnostic_source(&self) -> Option<&dyn Diagnostic> {
-        Some(&self.kind as &dyn Diagnostic)
+        match &self.additional_context {
+            Some(ctx) => Some(ctx as &dyn Diagnostic),
+            None => None,
+        }
```

Comment:
At this point, you don't need to match anymore
```suggestion
        self.additional_context.as_ref().map(|ctx| ctx as &dyn Diagnostic)
```

Reply:
Resolved by 1954a299b